lib/core/connector_job.rb

```diff
@@ -111,6 +118,33 @@ def reload
       @connector = nil
     end
 
+    def with_lock
+      seq_no = nil
+      primary_term = nil
+
+      response = ElasticConnectorActions.get_job(id)
+
+      yield response, seq_no, primary_term
+    end
+
+    def process!
```

Comment:
So it's the actual job claiming, right? It does not actually run the sync?

Reply:
Yeah, it changes the job statuses. The method naming comes from https://github.com/state-machines/state_machines which might be a bit confusing. After your question, I also find it weird. 

Do you think I can rename it to something like `job.make_running!` or `job.running!`. open for any ideas :)

Reply:
We agreed on `job.make_running!`